### DISCUSSION TITLE
Waveshare RS485 to ETH module

### DISCUSSION THREAD

**Discussion by ermagan** [omerfaruk-aran/esphome_samsung_hvac_bus#14] [General]
Hello Ianwin/Steve,

Firstly, thank you for creating this. We want to implement the module for a Samsung EHS Heatpump.

You are currently working on this beautiful script with an ESP module. We were wondering if it is possible to also work with a Waveshare RS485 to ETH module. This module establishes a TCP connection with a host/IP address to read and write via the RS485 ports. Here is a [video](https://www.youtube.com/watch?v=Xuj2YFZ5zME) of someone who is working with the module.

Is it maybe also possible for you to create a simple manual for the integration with Home Assistant?

Best regards

**Comment by lanwin** [comment]
Hi, technically the NASA parsing code should also work when data comes via TCP instead directly via Serial from RS485. 

But the way this module here is build, focuses directly on the ESPHome UART component (Serial). Maybe its possible to extend it to receive data also via TCP, but I did not think that this will be added in the near future...simply cause my time is pretty limited for now. 

If someone wants to impl it. I am available to discuss how that could work. And also happy to help someone write a pullrequest.

**Comment by bzumik1** [comment]
Hi ermagan, could you please explain me the advantages of using this instead of ESP board? If you want to connect the Heatpump over ethernet there should be ESP board with ethernet. There is even this prebuilt desk from M5Stack https://shop.m5stack.com/products/lan-module-with-w5500-v12

**Comment by lanwin** [comment]
Btw. If you dont want to have wifi. There are also ESP boards out there with Ethernet support.

**Comment by ermagan** [comment]
Thank you for your response. We do have the module available, which is why we were interested in its compatibility. We thought that perhaps it could be a simple configuration adjustment in the .yaml file, such as setting the type to 'tcp', host to 'xxx.xxx.xxx.xxx', and port to '502', instead of using the UART configuration.

**Comment by ermagan** [comment]
We have ordered the M5STACK Atom Lite + RS485. Waiting for delivery. As I mentioned earlier, we want to read the Samsung heat pump based on the NASA protocol. We have already made an attempt to read the source code of Samsung's S-NET pro 2 and looked for commonalities in the repository on how you used it for the AC. This concerns the EHS Mono system heat pump from Samsung. Is there a possibility for us to get in direct contact with you? Perhaps via WhatsApp/Telegram?

**Reply by lanwin** [reply]
Why did you think we need direct contact? Personally I would appreciate to discuss this in public so everyone can participate.

**Comment by Mocky158** [comment]
I see that you want to communicate with the same heat pump as me.
I have a question whether the ESP should be connected to the outdoor or indoor unit?
I think it should not matter since they are also connected via the F1/F2 connector if I'm not mistaken.
But I was shocked by what I found in the manual and what is written on page 5.

[https://s3.amazonaws.com/samsung-files/Tech_Files/SNET+Pro+and+SNET+Pro+2+Service+Software/Snet+Pro+Service+Software_SNET+Pro+2+Instructions.pdf](url)

**Reply by lanwin** [reply]
No it did not matter where you connect it. The connection between the devices acts like bus. Every device witch is connected receives all messages.
